documentation/src/developer_documentation/state_migration_spike.md

```diff
@@ -0,0 +1,139 @@
+## State migration spike 🛂
+
+### What is state migration?
+
+State migration is a process where the `StateTree` contents are transformed from
+an older form to a newer form.
+
+### Why do we need to migrate?
+
+Migration is required when the structure of the state changes. This happens when
+new fields are added or existing ones are modified. Migration is **not**
+required in case of new behaviour.
+
+In case of NV18, the `StateTree` changed from version 4 to version 5. See
+https://github.com/filecoin-project/ref-fvm/pull/1062
+
+### What to upgrade?
+
+We need to upgrade the `StateTree` which is represented as
```

Comment:
Same as comment above. It's more than just upgrading the StateTree for NV18 it seems.